src/osd/PrimaryLogPG.cc

```diff
@@ -8504,9 +8504,16 @@ void PrimaryLogPG::_do_rollback_to(OpContext *ctx, ObjectContextRef rollback_to,
   map<snapid_t, interval_set<uint64_t> >::iterator iter =
     snapset.clone_overlap.lower_bound(snapid);
   ceph_assert(iter != snapset.clone_overlap.end());
-  interval_set<uint64_t> overlaps = iter->second;
+  auto& last_clone_iter = --snapset.clone_overlap.end();
```

Comment:
auto & is wrong here -- it's capturing a reference to a temporary.

Reply:
Ah, that should have been a compilation error but the `--` hides it.